### DISCUSSION TITLE
What is the URL -- what's on 8095 - and how do i get it?

### DISCUSSION THREAD

**Discussion by Br3nda** [music-assistant/support#2016] [Q&A] [answered]
I'm trying to install Music Assistant -- but I'm struggling with the docs

I think it's saying I don't need to install the server; I just install the HA addon and that provides the server. Is that right? The docs seem to imply which but without saying it, using the title "Primary installation method: Home Assistant Integration Installation"

I have added the integration using HACS, and restarted HASS (I rebooted the whole machine in desperation)

I'm now stuck here "Go to HA Settings>>Devices & services>>Integrations and click the big + ADD INTEGRATION button."

The dialogue appears with "The URL of the Music Assistant server*" and is pre-filled with `http://mass.local:8095`. I tried that, and I got `"Failed to Connect"` I can see that `mass.local` doesn't resolve, so i changed to localhost.  -- but i can see port 8095 localhost is connection refused. There's nothing there. Eveything i try gets `"Failed to Connect"`

```
$ telnet localhost 8095
Trying 127.0.0.1...
telnet: Unable to connect to remote host: Connection refused
```

So, now I'm wondering if there's some docs that i have missed entirely, and i need to separately go install some server daemon that will end up listening on post 8095.. but the docs says "Our recommended installation method is to run the server as a Home assistant Add-on which will be automatically installed when you add the Home Assistant Integration"

So, I'm lost. What do I put in that dialogue? What's supposed to be on port 8095? How do I get it?

![Screenshot from 2024-03-17 09-43-48](https://github.com/music-assistant/hass-music-assistant/assets/12266/5f2f591a-5404-4704-aabe-c979d07f1481)

**Comment by OzGav** [comment]
Are you using HAOS?

**Reply by Br3nda** [reply]
i don't know what that is, so I guess no?

**Reply by Br3nda** [reply]
i have Homeassistant from pip, running on an old laptop, on ubuntu

**Comment by Br3nda** [comment]
got it working by running the music assistant docker

the wording in the docs made me think all I needed was the Home Assistant Add on. the way it says "Primary installation method", "Secondary...", "Tertiary.."" sound like you prefer the first and then secondary and tertiary are other ways of doing it, but I guess you need to do all three. 
https://music-assistant.io/installation/

**Reply by OzGav** [reply]
The first words of both the primary and secondary method is `Using HAOS`If you didn’t know what that was then you could have googled it and it is the first result. Sounds like you have it working now though so all good.

**Reply by kanesbit** [reply, marked as answer]
I have this same question.  but have figured it out.

See this tip from the docco.

_You can keep the server more secure by NOT exposing the webserver and let the addon talk directly to the webserver on the internal docker network. In that case address the internal dns name of the addon would be, for example, http://d5369777-music-assistant-beta:8095/_

I am running HAOS, on Proxmox, installed using the proxmox helper scripts. 

My read of the documentation tells me you don't have to do all 3.  They are the order of preference and ease, so just install it as HAOS, automatically as an addon
